business/istio_validations.go

```diff
@@ -480,6 +473,62 @@ func (in *IstioValidationsService) fetchDetails(rValue *kubernetes.IstioDetails,
 	}
 }
 
+func (in *IstioValidationsService) fetchAllVirtualServices(istioDetails *kubernetes.IstioDetails, namespace string, errChan chan error, wg *sync.WaitGroup) {
+	defer wg.Done()
+	if len(errChan) == 0 {
+		nss, err := in.businessLayer.Namespace.GetNamespaces()
+		if err != nil {
+			errChan <- err
+			return
+		}
+
+		currentVSs, err := in.fetchVirtualServices(namespace)
+		if err != nil {
+			select {
+			case errChan <- err:
+			default:
+			}
+		} else {
+			for _, ns := range nss {
+				vsList, err := in.fetchVirtualServices(ns.Name)
+				if err != nil {
+					select {
+					case errChan <- err:
+					default:
+					}
+				} else {
+					for _, vs := range currentVSs {
+						var namespaceFound = false
+						if exportToSpec, found := vs.GetSpec()["exportTo"]; found {
+							if namespaces, ok := exportToSpec.([]interface{}); ok {
+								for _, exportToNs := range namespaces {
+									// take only namespaces where it is exported to, or if it is exported to current namespace
+									if (exportToNs == "." && ns.Name == vs.GetObjectMeta().Namespace) || exportToNs == ns.Name || exportToNs == "*" {
+										namespaceFound = true
+									}
+								}
+							}
+						} else {
+							namespaceFound = true
+						}
+						if namespaceFound {
+							istioDetails.VirtualServices = append(istioDetails.VirtualServices, vsList...)
+						}
+					}
+				}
+			}
+		}
+	}
+}
```

Comment:
One thing we should consider is to add test to this enhancement. Sooner than later. I guess that thinking the tests upfront would help you better understand the feature from the bottom to the top. For example, add a validation for virtual services that fails when there is one virtual service outside of the ns but exported to that ns.